src/git/Repository.ts

```diff
@@ -223,6 +223,18 @@ export class Repository {
         });
     }
 
+    public getLatestTagOnBranch(tagPattern: string, branch: string): Promise<string> {
+        return new Promise<string[]>((resolve, reject) => {
+            this.git.raw(['describe', '--abbrev=0', `--match=${tagPattern}`, branch], (err, result: string) => {
```

Comment:
This could be solved with a remote call to the repo which would not require the fetch above:
`git ls-remote --tags --sort="version:refname" git@github.com:collaborationFactory/${repo}.git "version/22.2.*"`

It can be filtered with wildcards etc. -> see `https://git-scm.com/docs/git-ls-remote.html`. 
If it is required to fetch the tag, maybe this wold work: `git fetch origin refs/tags/1.0.0` but on circleci we rather should clone the tag directly.

Reply:
As I said: the above code would be useful for local development, so maybe we need to different ways...